composition/content_fetcher.go

```diff
@@ -115,9 +129,24 @@ func (fetcher *ContentFetcher) AddFetchJob(d *FetchDefinition) {
 
 		if fetchResult.Err == nil {
 			fetcher.addMeta(fetchResult.Content.Meta())
-			for _, dependency := range fetchResult.Content.RequiredContent() {
-				if dependency.IsFetchable() {
-					fetcher.AddFetchJob(dependency)
+			for _, fetch := range fetchResult.Content.RequiredContent() {
+				fetcher.AddFetchJob(fetch)
+			}
+			for dependencyName, params := range fetchResult.Content.Dependencies() {
+				_, exist := fetcher.r.sheduledFetchDefinitionNames[dependencyName]
+				if !exist {
+					lazyFd, exist, err := fetcher.lazyFdFactory(dependencyName, params)
+					if err != nil {
+						logging.Logger.WithError(err).
+							WithField("dependencyName", dependencyName).
+							WithField("params", params).
+							Errorf("failed fetching dependency %v", dependencyName)
```

Comment:
The error message is telling the wrong thing. The dependency ist not being fetched here. The error should be something like "failed creating fetchdefinition for dependency..."